### PR TITLE
:sparkles: :building_construction: Config editor

### PR DESCRIPTION
**Pull request by Tugduoff**
Added a config editor program. Look by yourself, its quite fun to use :)

(This program will work when second game pr is pushed since we need /config/levels to be present and its created there.)

### CODE REVIEW

File: configEditor/gui_window_file_dialog.h [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+        strcpy(state.fileNameText, GetFileName(initPath));
+    }
+    else strcpy(state.dirPathText, GetWorkingDirectory());
+
+    // TODO: Why we keep a copy?
+    strcpy(state.dirPathTextCopy, state.dirPathText);
+    strcpy(state.fileNameTextCopy, state.fileNameText);
+
+    state.filterExt[0] = '\0';
+    //strcpy(state.filterExt, "all");
```
Comment (Tholarr) [inline configEditor/gui_window_file_dialog.h:214]: comment left on purpose ?

File: configEditor/gui_window_file_dialog.h [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+        // Load current directory files
+        if (state->dirFiles.paths == NULL) ReloadDirectoryFiles(state);
+        //----------------------------------------------------------------------------------------
+
+        // Draw window and controls
+        //----------------------------------------------------------------------------------------
+        state->windowActive = !GuiWindowBox(state->windowBounds, "#198# Select File Dialog");
+
+        // Draw previous directory button + logic
+        // if (GuiButton((Rectangle){ state->windowBounds.x + state->windowBounds.width - 48, state->windowBounds.y + 24 + 12, 40, 24 }, "< .."))
```
Comment (Tholarr) [inline configEditor/gui_window_file_dialog.h:278]: comments left on purpose ? (line 278 to 308)

File: configEditor/gui_window_file_dialog.h [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+# else
+        GuiListViewEx((Rectangle){ state->windowBounds.x + 8, state->windowBounds.y + 10 + 20, state->windowBounds.width - 16, state->windowBounds.height - 22 - 16 - 68 }, 
+                      (const char**)dirFilesIcon, state->dirFiles.count, &state->filesListScrollIndex, &state->filesListActive, &state->itemFocused);
+# endif
+        GuiSetStyle(LISTVIEW, TEXT_ALIGNMENT, prevTextAlignment);
+        GuiSetStyle(LISTVIEW, LIST_ITEMS_HEIGHT, prevElementsHeight);
+
+        // Check if a path has been selected, if it is a directory, move to that directory (and reload paths)
+        if ((state->filesListActive >= 0) && (state->filesListActive != state->prevFilesListActive))
+            //&& (IsMouseButtonPressed(MOUSE_LEFT_BUTTON) || IsKeyPressed(KEY_ENTER) || IsKeyPressed(KEY_DPAD_A)))
```
Comment (Tholarr) [inline configEditor/gui_window_file_dialog.h:326]: comment left on purpose ?

File: configEditor/raygui.h [hunk 4]
```diff
@@ -0,0 +1,5521 @@
...
+    // NOTE: Cursor moves automatically when key down after some time
+    if (IsKeyDown(KEY_LEFT) || IsKeyDown(KEY_RIGHT) || IsKeyDown(KEY_UP) || IsKeyDown(KEY_DOWN) || IsKeyDown(KEY_BACKSPACE) || IsKeyDown(KEY_DELETE)) autoCursorCooldownCounter++;
+    else
+    {
+        autoCursorCooldownCounter = 0;      // GLOBAL: Cursor cooldown counter
+        autoCursorDelayCounter = 0;         // GLOBAL: Cursor delay counter
+    }
+
+    // Blink-cursor frame counter
+    //if (!autoCursorMode) blinkCursorFrameCounter++;
```
Comment (Tholarr) [inline configEditor/raygui.h:2486]: comments left on purpose ?

File: configEditor/raygui.h [hunk 5]
```diff
@@ -0,0 +1,5521 @@
...
+                    keyCount--;
+                    textValue[keyCount] = '\0';
+                    valueHasChanged = true;
+                }
+            }
+
+            if (valueHasChanged) *value = TextToInteger(textValue);
+
+            // NOTE: We are not clamp values until user input finishes
+            //if (*value > maxValue) *value = maxValue;
```
Comment (Tholarr) [inline configEditor/raygui.h:2888]: comments left on purpose ?

File: configEditor/raygui.h [hunk 6]
```diff
@@ -0,0 +1,5521 @@
...
+        }
+        else
+        {
+            if (((startIndex + i) == itemSelected) && (active != NULL))
+            {
+                // Draw item selected
+                GuiDrawRectangle(itemBounds, GuiGetStyle(LISTVIEW, BORDER_WIDTH), GetColor(GuiGetStyle(LISTVIEW, BORDER_COLOR_PRESSED)), GetColor(GuiGetStyle(LISTVIEW, BASE_COLOR_PRESSED)));
+                GuiDrawText(text[startIndex + i], GetTextBounds(DEFAULT, itemBounds), GuiGetStyle(LISTVIEW, TEXT_ALIGNMENT), GetColor(GuiGetStyle(LISTVIEW, TEXT_COLOR_PRESSED)));
+            }
+            else if (((startIndex + i) == itemFocused)) // && (focus != NULL))  // NOTE: We want items focused, despite not returned!
```
Comment (Tholarr) [inline configEditor/raygui.h:3301]: first part of the comment left on purpose ?

File: configEditor/raygui.h [hunk 7]
```diff
@@ -0,0 +1,5521 @@
...
+            if (IsMouseButtonDown(MOUSE_LEFT_BUTTON))
+            {
+                state = STATE_PRESSED;
+                guiSliderDragging = true;
+                guiSliderActive = bounds; // Store bounds as an identifier when dragging starts
+
+                *alpha = (mousePoint.x - bounds.x)/bounds.width;
+                if (*alpha <= 0.0f) *alpha = 0.0f;
+                if (*alpha >= 1.0f) *alpha = 1.0f;
+                //selector.x = bounds.x + (int)(((alpha - 0)/(100 - 0))*(bounds.width - 2*GuiGetStyle(SLIDER, BORDER_WIDTH))) - selector.width/2;
```
Comment (Tholarr) [inline configEditor/raygui.h:3478]: comment left on purpose ?

File: configEditor/raygui.h [hunk 8]
```diff
@@ -0,0 +1,5521 @@
...
+                guiSliderActive = bounds; // Store bounds as an identifier when dragging starts
+
+                *hue = (mousePoint.y - bounds.y)*360/bounds.height;
+                if (*hue <= 0.0f) *hue = 0.0f;
+                if (*hue >= 359.0f) *hue = 359.0f;
+
+            }
+            else state = STATE_FOCUSED;
+
+            /*if (IsKeyDown(KEY_UP))
```
Comment (Tholarr) [inline configEditor/raygui.h:3568]: comments left on purpose ? (line 3568 to 3577)

File: configEditor/raygui.h [hunk 9]
```diff
@@ -0,0 +1,5521 @@
...
+{
+    int result = 0;
+
+    Color temp = { 200, 0, 0, 255 };
+    if (color == NULL) color = &temp;
+
+    GuiColorPanel(bounds, NULL, color);
+
+    Rectangle boundsHue = { (float)bounds.x + bounds.width + GuiGetStyle(COLORPICKER, HUEBAR_PADDING), (float)bounds.y, (float)GuiGetStyle(COLORPICKER, HUEBAR_WIDTH), (float)bounds.height };
+    //Rectangle boundsAlpha = { bounds.x, bounds.y + bounds.height + GuiGetStyle(COLORPICKER, BARS_PADDING), bounds.width, GuiGetStyle(COLORPICKER, BARS_THICK) };
```
Comment (Tholarr) [inline configEditor/raygui.h:3622]: comment left on purpose ?

File: configEditor/raygui.h [hunk 10]
```diff
@@ -0,0 +1,5521 @@
...
+
+    Vector3 hsv = ConvertRGBtoHSV(RAYGUI_CLITERAL(Vector3){ (*color).r/255.0f, (*color).g/255.0f, (*color).b/255.0f });
+
+    GuiColorBarHue(boundsHue, NULL, &hsv.x);
+
+    //color.a = (unsigned char)(GuiColorBarAlpha(boundsAlpha, (float)color.a/255.0f)*255.0f);
```
Comment (Tholarr) [inline configEditor/raygui.h:3628]: comment left on purpose ?

**Review by Tholarr — CHANGES_REQUESTED** [review]
I didn't understand everything, but nothing jumped out at me. I'm now waiting for answers to the questions I asked and that'll be all right :+1:

**Review by Tholarr — APPROVED** [review]

**Review by C0Florent — APPROVED** [review]

**Pull request merged**